spec/models/solidus_volume_pricing/pricer_spec.rb

```diff
@@ -0,0 +1,280 @@
+        it 'uses the volume price' do
+          is_expected.to eq('$9.00')
+        end
+
+        context 'when volume price has a role' do
+          before do
+            variant.volume_prices.first.update(role_id: role.id)
+          end
+
+          context 'when no user is given' do
+            it_behaves_like 'having the variant price'
+          end
+
+          context 'when a user is given' do
+            let(:pricing_options) do
+              SolidusVolumePricing::PricingOptions.new(
+                quantity: quantity,
+                user: user
+              )
+            end
+
+            context 'whose role matches' do
+              before { user.spree_roles << role }
+
+              it 'uses the volume price' do
+                is_expected.to eq('$9.00')
+              end
+            end
+
+            context 'whose role does not match' do
+              before { user.spree_roles << other_role }
+
+              it_behaves_like 'having the variant price'
+            end
+          end
+        end
+      end
+
+      context 'when it does match from a volume price model' do
+        let(:quantity) { 6 }
+
+        before do
+          variant.volume_price_models << create(:volume_price_model)
+          variant.volume_price_models.first.volume_prices.create!(amount: 2, discount_type: 'dollar', range: '(5+)')
+        end
+
+        it 'uses the volume price from model' do
+          is_expected.to eq('$8.00')
+        end
+      end
+    end
+
+    context 'discount_type = percent' do
+      before do
+        variant.volume_prices.create!(amount: 0.25, discount_type: 'percent', range: '(10+)')
+      end
+
+      context 'when quantity does not match the range' do
+        it_behaves_like 'having the variant price'
+      end
+
+      context 'when quantity matches the range' do
+        let(:quantity) { 10 }
+
+        it 'uses the volume price' do
+          is_expected.to eq('$7.50')
+        end
+
+        context 'when volume price has a role' do
+          before do
+            variant.volume_prices.first.update(role_id: role.id)
+          end
+
+          context 'when no user is given' do
+            it_behaves_like 'having the variant price'
+          end
+
+          context 'when a user is given' do
+            let(:pricing_options) do
+              SolidusVolumePricing::PricingOptions.new(
+                quantity: quantity,
+                user: user
+              )
+            end
+
+            context 'whose role matches' do
+              before { user.spree_roles << role }
+
+              it 'uses the volume price' do
+                is_expected.to eq('$7.50')
+              end
+            end
+
+            context 'whose role does not match' do
+              before { user.spree_roles << other_role }
+
+              it_behaves_like 'having the variant price'
+            end
+          end
+        end
+      end
+
+      context 'when it does match from a volume price model' do
```

Comment:
Yeah, I was actually taking the original spec names that where already there. I will rename it into something understandable.